src/lib.rs

```diff
@@ -82,12 +82,23 @@ fn run_parser<'t>(
     payload: &str,
     output_count: &mut i32,
     output: &mut Vec<(PathBuf, String)>,
-    compile_directory: &mut Vec<(String, String, i32)>,
+    compile_directory: &mut Vec<(String, String, i32, String)>,
     multi: &MultiProgress,
     stats: &mut Stats,
 ) {
     if let Some(md) = parser.get_metadata(&e) {
         let results = parser.parse(lineno, md, e.rank, &e.compile_id, &payload);
+        fn extact_suffix(filename: &String) -> String {
+            if filename.contains("fx_graph_cache_miss") {
+                "❌".to_string()
+            } else if filename.contains("fx_graph_cache_hit") {
+                "✅".to_string()
+            } else if filename.contains("fx_graph_cache_bypass") {
+                "❓".to_string()
```

Comment:
So on the one hand, I don't have a huge problem with just hacking in UI stuff when we need to (indeed, the original version of the code was just one giant for loop with all the logic in one places). @jamesjwu refactored things so that we had separate parsers for different things we might want to do. In James's model, this logic should be payload agnostic. But it clearly is not here, you're adding special case for certain file types. The other model is that you have to write a dedicated parser for this. The code in tlparse right now isn't really structured this way, but my vision for the future is that we have a default parser (this) for regular non-artifact entries, but then you can write custom parsers when you need to customize (so this would be a custom parser).

So I am meh. I'll give a stamp. But if @jamesjwu does have an opinion I'm inclined to listen to him.